crates/router/src/connector/bankofamerica/transformers.rs

```diff
@@ -63,105 +192,290 @@ impl TryFrom<&BankofamericaRouterData<&types::PaymentsAuthorizeRouterData>>
+                        total_amount: item.amount.to_owned(),
+                        currency: item.router_data.request.currency.to_string().to_uppercase(),
+                    },
+                    bill_to,
+                };
+                let card_issuer = ccard.get_card_issuer();
+                let card_type = match card_issuer {
+                    Ok(issuer) => Some(get_card_type(issuer).to_string()),
+                    Err(_) => None,
                 };
+                let payment_information = PaymentInformation {
+                    card: Card {
+                        number: ccard.card_number,
+                        expiration_month: ccard.card_exp_month,
+                        expiration_year: ccard.card_exp_year,
+                        security_code: ccard.card_cvc,
+                        card_type,
+                    },
+                };
+
+                let processing_information = ProcessingInformation {
+                    capture: matches!(
+                        item.router_data.request.capture_method,
+                        Some(enums::CaptureMethod::Automatic) | None
+                    ),
+                    capture_options: None,
+                };
+
+                let client_reference_information = ClientReferenceInformation {
+                    code: Some(item.router_data.connector_request_reference_id.clone()),
+                };
+
                 Ok(Self {
-                    amount: item.amount.to_owned(),
-                    card,
+                    processing_information,
+                    payment_information,
+                    order_information,
+                    client_reference_information,
                 })
             }
             _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
-// Auth Struct
-pub struct BankofamericaAuthType {
-    pub(super) api_key: Secret<String>,
-}
-
-impl TryFrom<&types::ConnectorAuthType> for BankofamericaAuthType {
-    type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
-        match auth_type {
-            types::ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
-                api_key: api_key.to_owned(),
-            }),
-            _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
-        }
-    }
-}
-// PaymentsResponse
-//TODO: Append the remaining status flags
-#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
-#[serde(rename_all = "lowercase")]
+#[derive(Debug, Default, Clone, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
 pub enum BankofamericaPaymentStatus {
+    Authorized,
     Succeeded,
     Failed,
+    Voided,
+    Reversed,
+    Pending,
+    Declined,
+    AuthorizedPendingReview,
+    Transmitted,
     #[default]
```

Comment:
why are we having default status?

Reply:
Because the documentation is not proper and we cannot be sure of all the possible values. Hence to avoid 5xx default has been introduced.

Reply:
removed